### PR TITLE
Add KFServing TF mnist S3 saved model example

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:
Add a TF mnist S3 model serving example to demonstrate how user can use `KFServing` to serve a model saved on S3.

**Special notes for your reviewer**:

If this example looks good we can mostly replicate this to create another example for GCS.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/112)

### CODE REVIEW

File: docs/samples/s3/README.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+export S3_ENDPOINT=minio-service.kubeflow:9000
+export AWS_ENDPOINT_URL=http://${S3_ENDPOINT}
+
+kustomize edit add configmap mnist-map-training --from-literal=S3_ENDPOINT=${S3_ENDPOINT}
+kustomize edit add configmap mnist-map-training --from-literal=AWS_ENDPOINT_URL=${AWS_ENDPOINT_URL}
+kustomize edit add configmap mnist-map-training --from-literal=S3_USE_HTTPS=${S3_USE_HTTPS}
+
+kustomize edit add configmap mnist-map-training --from-literal=modelDir=s3://mnist/v1
+kustomize edit add configmap mnist-map-training --from-literal=exportDir=s3://mnist/v1/export
+```
```
Comment (animeshsingh) [inline docs/samples/s3/README.md]: We need to simplify the user experience. Assuming I as a user have specified the two yamls, I should not be required to do anything with kustomize?

File: docs/samples/s3/README.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: default # change to your service account name
+secrets:
+- name: default-token-zjpsv
+- name: mysecret # append the s3 secret here and save
+```
```
Comment (animeshsingh) [inline docs/samples/s3/README.md]: Same here - this is offloading and exposing too much of kube backend to end users?

File: docs/samples/s3/s3_secret.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
```
Comment (ellis-bigelow) [inline docs/samples/s3/s3_secret.yaml]: Should we provide yaml for the service account?

File: docs/samples/s3/s3_secret.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+kind: Secret
+metadata:
+  name: mysecret
+  annotations:
+     serving.kubeflow.org/s3-endpoint: minio-service.kubeflow:9000 # replace with your s3 endpoint
```
Comment (ellis-bigelow) [inline docs/samples/s3/s3_secret.yaml]: Is minio-service already deployed somewhere in some kubeflow installations?

File: docs/samples/s3/README.md (hunk 1 above)
Comment (ellis-bigelow) [reply docs/samples/s3/README.md]: Agreed.

Comment (yuzisun) [reply docs/samples/s3/README.md]: these settings are actually for the kubeflow TF training job to save the model,  are you saying we directly provide the TF job yaml ? This step is not required if you already have a model on S3.

File: docs/samples/s3/README.md (hunk 2 above)
Comment (yuzisun) [reply docs/samples/s3/README.md]: Any thoughts how to improve the experience ? We need some way to create secret and attach the secret to service account.

File: docs/samples/s3/s3_secret.yaml (hunk 4 above)
Comment (yuzisun) [reply docs/samples/s3/s3_secret.yaml]: yes that's part of the kubeflow installation.

File: docs/samples/s3/s3_secret.yaml (hunk 3 above)
Comment (yuzisun) [reply docs/samples/s3/s3_secret.yaml]: yes we can do that

File: docs/samples/s3/README.md (hunk 1 above)
Comment (yuzisun) [reply docs/samples/s3/README.md]: I think I can just add the actual TF job yaml without above steps, but trust me the yaml looks a bit scary :)

Comment (animeshsingh) [reply docs/samples/s3/README.md]: If I have not saved a model on s3, the steps would be more like
1. Use s3 or minio client to save your model on s3
2. Add your s3 location in yaml
3. Apply yaml.

Then we have automation to take the yaml and do what is needed in background?

Why the user is being asked to manipulate configmap settings? Maybe this is point in time, but this needs to be improved?

File: docs/samples/s3/README.md (hunk 2 above)
Comment (animeshsingh) [reply docs/samples/s3/README.md]: Ok - so I was discussing with @ellis-bigelow to move to "One" yaml semantic, and possibly align with MLSPec. Given that this is how currently things are, we shall get this PR in. I will bring it up on our next weekly sync what would be ideal experience there.

File: docs/samples/s3/README.md (hunk 1 above)
Comment (yuzisun) [reply docs/samples/s3/README.md]: Chatted with Animesh offline, this is the current way how kubeflow does for the TF training job, we can change to use arena cli for better user experience.

**Comment by animeshsingh** [comment]
Given this is a useful example sample showing how to get things done with current code-base, we should get this in, and then evolve it to work towards better user-experience

/lgtm   cc @ellis-bigelow

**Comment by animeshsingh** [comment]
//lgtm

File: docs/samples/s3/tensorflow_s3.yaml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+apiVersion: "serving.kubeflow.org/v1alpha1"
+kind: "KFService"
+metadata:
+  name: "mnist-s3"
+spec:
+  default:
+    tensorflow:
+      # serviceAccountName: sa   // overwrite default service account
```
Comment (ellis-bigelow) [inline docs/samples/s3/tensorflow_s3.yaml]: Can you delete this?

Comment (ellis-bigelow) [reply docs/samples/s3/tensorflow_s3.yaml]: Rather, can we wire it up such that the service account in the docs points directly to this?

Comment (yuzisun) [reply docs/samples/s3/tensorflow_s3.yaml]: Done

**Pull request merged**